### PR TITLE
feat(auth): secure admin api with hmac signatures

### LINKED ISSUE #2704: Secure Auth Server Admin API with signatures
As was done with the [backend server's admin API](https://github.com/interledger/rafiki/pull/2632), the auth server's admin API needs to be secured in the same way.

### PR DESCRIPTION
**Pull request by njlie**
## Changes proposed in this pull request

- Added signature header verification to Auth Server Admin API

## Context

Closes #2704.

## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass
- [x] Bruno collection updated

### CODE REVIEW

File: packages/auth/src/config/app.ts [hunk 1]
```diff
@@ -45,6 +45,8 @@ export const Config = {
     'AUTH_SERVER_DOMAIN',
     `http://localhost:${envInt('AUTH_PORT', 3006)}`
   ),
+  apiSecret: process.env.API_SECRET, // optional
+  apiSignatureVersion: envInt('API_SIGNATURE_VERSION', 1),
```
Comment (mkurapov) [inline packages/auth/src/config/app.ts:49]: Not related to this PR necessarily, but what is the benefit of having version? just to track how many times a key has been rotated, basically?

Also, should we call this "Admin API secret"?

File: packages/auth/src/shared/utils.ts [hunk 2]
```diff
@@ -7,3 +10,28 @@ export function generateNonce(): string {
...
+
+export function verifyApiSignature(ctx: Context, config: IAppConfig): boolean {
+  const { headers, body } = ctx.request
+  const signature = headers['signature']
+  if (!signature) {
+    return false
+  }
+
+  const signatureParts = (signature as string)?.split(', ')
+  const timestamp = signatureParts[0].split('=')[1]
```
Comment (golobitch) [inline packages/auth/src/shared/utils.ts:22]: Replay attack? So nothing here prevents us to get the same and same request over and over again, correct. As long that once it was correct, we can get it n number of times.

I would do it like this:
1. Save the signature to redis for example with key: `signature:${headers['signature']}`
2. On line 20 check if this key exists. If exists, return Unauthorized
3. Check the validity of request ... basically...how long after the request has been signed, we can still process this request? Let's say 30 seconds? Or even less? Also note, that this value is also the TTL of the above signature key

File: packages/auth/src/config/app.ts (hunk 1 above)
Comment (njlie) [reply packages/auth/src/config/app.ts:49]: Not sure what the answer to having version is, but I'm down to rename the env var

File: packages/auth/src/shared/utils.ts (hunk 2 above)
Comment (njlie) [reply packages/auth/src/shared/utils.ts:22]: Need to do this on the `backend` package as well. Going to capture in https://github.com/interledger/rafiki/issues/2725

File: packages/auth/src/shared/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  let redis: Redis
+
+  describe('admin api signatures', (): void => {
+    beforeAll(async (): Promise<void> => {
+      deps = initIocContainer({
+        ...Config,
+        adminApiSecret: 'test-secret'
+      })
+      appContainer = await createTestApp(deps)
+      const redis = await deps.use('redis')
```
Comment (BlairCurrey) [inline packages/auth/src/shared/utils.test.ts:24]: (doesn't fix the `MaxRetriesPerRequestError` error)

```suggestion
     redis = await deps.use('redis')
```

File: packages/auth/jest.setup.js [hunk 4]
```diff
@@ -66,4 +67,12 @@ module.exports = async (globalConfig) => {
   }
 
   global.__AUTH_KNEX__ = db
+  if (!process.env.REDIS_URL) {
+    const redisContainer = await new GenericContainer('redis:7')
+      .withExposedPorts(REDIS_PORT)
+      .start()
+
+    global.__AUTH_REDIS__ = redisContainer
+    process.env.REDIS_URL = `redis://localhost${redisContainer.getMappedPort(REDIS_PORT)}`
```
Comment (BlairCurrey) [inline packages/auth/jest.setup.js:76]: missing colon

```suggestion
    process.env.REDIS_URL = `redis://localhost:${redisContainer.getMappedPort(REDIS_PORT)}`
```

**Review by BlairCurrey — DISMISSED** [review]

File: packages/auth/src/shared/utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      const requestBody = { test: 'value' }
+      const signature = generateApiSignature(
+        'test-secret',
+        Config.adminApiSignatureVersion,
+        requestBody
+      )
+      const key = `signature:${signature}`
+      const op = redis.multi()
+      op.set(key, signature)
+      op.expire(key, signature)
```
Comment (mkurapov) [inline packages/auth/src/shared/utils.test.ts:90]: should the expiry take in a num as the second argument?

File: packages/auth/src/shared/utils.ts [hunk 6]
```diff
@@ -24,3 +27,90 @@ export function generateRouteLogs(ctx: AppContext): {
...
+  logger.info(
+    {
+      currentTime,
+      signatureTime,
+      diff: currentTime - signatureTime,
+      ttlMilliseconds
+    },
+    'time differential'
+  )
+  if (currentTime - signatureTime > ttlMilliseconds) return false
```
Comment (mkurapov) [inline packages/auth/src/shared/utils.ts:88]: I think we should add a test for this as well

**Review by BlairCurrey — APPROVED** [review]

**Review by mkurapov — APPROVED** [review]

**Pull request merged**